src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2948,23 +2970,7 @@ int RGWBucketInstanceMetadataHandler::put_post(
 int RGWBucketInstanceMetadataHandler::remove(std::string& entry, RGWObjVersionTracker& objv_tracker,
                                              optional_yield y, const DoutPrefixProvider *dpp)
 {
-  RGWBucketCompleteInfo bci;
-  int ret = svc_bucket->read_bucket_instance_info(entry, &bci.info, nullptr,
-                                                  &bci.attrs, y, dpp);
-  if (ret == -ENOENT) {
-    return 0;
-  }
-  if (ret < 0) {
-    return ret;
-  }
-
-  ret = svc_bucket->remove_bucket_instance_info(
-      entry, bci.info, &bci.info.objv_tracker, y, dpp);
-  if (ret < 0)
-    return ret;
-  std::ignore = update_bucket_topic_mappings(dpp, &bci, /*current_bci=*/nullptr,
-                                             driver);
```

Comment:
`RGWBucketInstanceMetadataHandler::remove()` defers the bucket instance removal until trim, but do we ever make this call to `update_bucket_topic_mappings()` later?

maybe we should keep the `read_bucket_instance_info()` and `update_bucket_topic_mappings()` calls here, and just skip the `remove_bucket_instance_info()`?

Reply:
yes this was a question I had. will add it back.